agent/consul/leader_connect_ca_test.go

```diff
@@ -689,6 +689,86 @@ func TestCAManager_Initialize_Vault_WithIntermediateAsPrimaryCA(t *testing.T) {
 	})
 }
 
+func TestCAManager_Verify_NoChangeToSecondaryConfig(t *testing.T) {
```

Comment:
Oh, I almost missed this!  It's unfortunate, but because of how our CI is setup right now, this test is not going to be run in CI.

To have it run properly we need the name to include the string `_Vault_`.  Otherwise the main suite skips this test (because the vault binary is not installed), and the "ca provider" test suite doesn't run the test because the name doesn't match.

It's really unfortunate that it's so easy to accidentally write a test that will never run. I've created #12466 to document this problem, and propose some options for fixing it.

For now, if you add `_Vault_` to the name, the test should run correctly.